pkg/reconciler/kubernetes/tektonchain/tektonchain.go

```diff
@@ -509,3 +513,41 @@ func (r *Reconciler) markUpgrade(ctx context.Context, tc *v1alpha1.TektonChain)
 	}
 	return v1alpha1.RECONCILE_AGAIN_ERR
 }
+
+func AddControllerEnv(controllerEnvs []v1alpha1.ControllerEnv) mf.Transformer {
+	return func(u *unstructured.Unstructured) error {
+		if u.GetKind() != "Deployment" {
```

Comment:
also include  `len(controllerEnvs) == 0`, so we can skip immediately, if there is no elements found in `controllerEnvs`
```
if u.GetKind() != "Deployment"  || len(controllerEnvs) == 0 {
```